pkg/proxy/kafka.go

```diff
@@ -159,13 +160,69 @@ func (k *kafkaRedirect) canAccess(req *kafka.RequestMessage, numIdentity policy.
 	return req.MatchesRule(rules.Kafka)
 }
 
+func (k *kafkaRedirect) localEndpointInfo(info *accesslog.EndpointInfo) {
+	LocalEndpointInfo(info, k.conf.source, k.epID)
+}
+
+// getInfoFromConsumable fills the accesslog.EndpointInfo fields, by fetching
+// the consumable from the consumable cache of endpoint using identity sent by
+// source. This is needed in ingress proxy while logging the source endpoint
+// info.  Since there will be 2 proxies on the same host, if both egress and
+// ingress policies are set, the ingress policy cannot determine the source
+// endpoint info based on ip address, as the ip address would be that of the
+// egress proxy i.e host.
+func (k *kafkaRedirect) getInfoFromConsumable(ipstr string,
+	info *accesslog.EndpointInfo,
+	srcIdentity policy.NumericIdentity) {
+	GetInfoFromConsumable(ipstr, info, srcIdentity, k.conf.source)
+}
+
+func (k *kafkaRedirect) getSourceInfo(remoteAddr string,
+	srcIdentity policy.NumericIdentity) (accesslog.EndpointInfo,
+	accesslog.IPVersion) {
+	return GetSourceInfo(remoteAddr, srcIdentity, k.ingress, k)
+}
+
+// fillReservedIdentity resolves the labels of the specified identity if known
+// locally and fills in the info member fields.
+func (k *kafkaRedirect) fillReservedIdentity(info *accesslog.EndpointInfo, id policy.NumericIdentity) {
+	FillReservedIdentity(info, id)
+}
+
+// egressDestinationInfo returns the destination EndpointInfo for a flow
+// leaving the proxy at egress.
+func (k *kafkaRedirect) egressDestinationInfo(ipstr string, info *accesslog.EndpointInfo) {
+	EgressDestinationInfo(ipstr, info)
+}
+
+// getDestinationInfo returns the destination EndpointInfo.
+func (k *kafkaRedirect) getDestinationInfo(dstIPPort string) accesslog.EndpointInfo {
+	return GetDestinationInfo(dstIPPort, k, k.ingress)
+}
+
 func (k *kafkaRedirect) handleRequest(pair *connectionPair, req *kafka.RequestMessage) {
 	scopedLog := log.WithField(fieldID, pair.String())
 	scopedLog.WithField(logfields.Request, req.String()).Debug("Handling Kafka request")
 
+	record := &accesslog.LogRecord{
+		KafkaRequest:      req,
+		Timestamp:         time.Now().UTC().Format(time.RFC3339Nano),
+		NodeAddressInfo:   k.nodeInfo,
+		TransportProtocol: 6, // TCP's IANA-assigned protocol number
+	}
+
+	if k.ingress {
+		record.ObservationPoint = accesslog.Ingress
+	} else {
+		record.ObservationPoint = accesslog.Egress
+	}
+
 	addr := pair.rx.conn.RemoteAddr()
 	if addr == nil {
 		scopedLog.Warn("RemoteAddr() is nil")
+		record.Info = fmt.Sprint("RemoteAddr() is nil")
```

Comment:
You can fill record first and then print the warning like this:
```golang
scopedLog.Warn(record.Info)
```

Reply:
done